politeiad/backendv2/tstorebe/store/mysql/mysql.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/decred/politeia/util"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+const (
+	// Database options
+	connTimeout     = 1 * time.Minute
+	connMaxLifetime = 1 * time.Minute
+	maxOpenConns    = 0 // 0 is unlimited
+	maxIdleConns    = 100
+
+	// Database table names
+	tableNameKeyValue = "kv"
+	tableNameNonce    = "nonce"
+)
+
+// tableKeyValue defines the key-value table.
+const tableKeyValue = `
+  k VARCHAR(38) NOT NULL PRIMARY KEY,
+  v LONGBLOB NOT NULL
+`
+
+// tableNonce defines the table used to track the encryption nonce.
+const tableNonce = `
+  n BIGINT PRIMARY KEY AUTO_INCREMENT
+`
+
+var (
+	_ store.BlobKV = (*mysql)(nil)
+)
+
+// mysql implements the store BlobKV interface using a mysql driver.
+type mysql struct {
+	db *sql.DB
+
+	// Encryption key and mutex. The key is zero'd out on application
+	// exit so the read lock must be held during concurrent access to
+	// prevent the golang race detector from complaining.
+	key    *[32]byte
+	keyMtx sync.RWMutex
```

Comment:
Move this to the front and make it unnamed.

Reply:
Done.

https://github.com/decred/politeia/pull/1180/files#diff-9cbdfdfae5a29aa39168b4484642d0acb8fdd880dcccdb358538928ac60cdb70R47